### ISSUE TITLE
Use SymEngine as a backend?

### ISSUE THREAD

**Issue by ChrisRackauckas** [jlapeyre/Symata.jl#29]
SymEngine is made by the same people as SymPy. It's a re-write of the engine to C++ and the benchmarks are fantastic. Although not currently feature-complete, you may want to start migrating over to this and get the speed benefits.

**Comment by jlapeyre** [comment]
I looked into it very briefly a while ago. It looks very promising. But, I wasn't able to find any documentation.

**Comment by ChrisRackauckas** [comment]
Sadly there isn't any yet: you have to look at their tests. But it is very good for what they have implemented.

**Comment by jlapeyre** [comment]
Yes, it takes a bit of poking around. Maybe the python bindings give more examples. I am in favor using existing libraries. I would consider symengine if it gains more features. Especially if it replaces the core of sympy, there is no reason to go through python.  But, the Symata interface to sympy and even Julia are not yet as complete and robust as I would like. So I haven't taken time yet to make try interfaces to libraries for number theory, etc.
